include/deal.II/lac/solver_gmres.h

```diff
@@ -630,30 +634,62 @@ class SolverGMRES : public SolverBase<VectorType>
 };
 
 
-
 /**
- * Implementation of the Generalized minimal residual method with flexible
- * preconditioning (flexible GMRES or FGMRES).
+ * Implementation of the multiply preconditioned generalized minimal
+ * residual method (MPGMRES).
  *
- * This flexible version of the GMRES method allows for the use of a different
- * preconditioner in each iteration step. Therefore, it is also more robust
- * with respect to inaccurate evaluation of the preconditioner. An important
- * application is the use of a Krylov space method inside the
- * preconditioner. As opposed to SolverGMRES which allows one to choose
- * between left and right preconditioning, this solver always applies the
- * preconditioner from the right.
+ * This method is a variant of the flexible GMRES, utilizing N
+ * preconditioners to search for a solution within a multi-Krylov space.
+ * These spaces are characterized by by all possible N-variate,
+ * non-commuting polynomials of the preconditioners and system matrix
+ * applied to a residual up to some fixed degree. In contrast, the flexible
+ * GMRES method implemented in SolverFGMRES constructs only one "Krylov"
+ * subspace, which is formed by univariate polynomials in one
+ * preconditioner that may change at each iteration.
  *
- * FGMRES needs two vectors in each iteration steps yielding a total of
- * <tt>2*SolverFGMRES::%AdditionalData::%max_basis_size+1</tt> auxiliary
- * vectors. Otherwise, FGMRES requires roughly the same number of operations
- * per iteration compared to GMRES, except one application of the
- * preconditioner less at each restart and at the end of solve().
+ * We implement two strategies, a "full" and a "truncated" MPGMRES version
+ * that differ in how they construct Krylov subspaces. The full MPGMRES
+ * version constructs a Krylov subspace for every possible combination of
+ * preconditioner application to the initial residual. For two
```

Comment:
If I read this and the example below correctly, then the number of combinations grows exponentially with the iteration count. This is also what the paper states in their Sec 2.4. I think it would make sense to state this behavior and the possible implications on the size of notion of iteration and costs. Maybe you can just refer to the reference and the tables in that paper?

Reply:
We have improved the `@note` in the class documentation that addresses this point.